File: BeeCrowd_C/1222.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* N = Número de palavras
 L = Número de linhas 
 C = Número de caracteres por linha */

 int main(void){
    int N, L, C;
    char entrada[100];
    char frase[100];
    int countCaracteres = 0;
    int countLinhas = 1;
    int countPags = 1;

    fgets(entrada, sizeof(entrada), stdin);

    //SSCANF SERVE PARA QUEBRAR UMA LINHA E ATRIBUIR VALORES A VARIAVEIS, COMO OCORRE COM O SPLIT NO PYTHON
    sscanf(entrada, "%d %d %d", &N, &L, &C);

    // LENDO A FRASE
    fgets(frase, sizeof(frase), stdin);

    for(int i = 0; i < strlen(frase); i++){
        // VERIFICANDO SE AINDA CABEM CARACTERES NA LINHA
        if (countCaracteres < C){
            countCaracteres++;
        }else{
            countCaracteres = 0;
            countLinhas++;
        }

        // RESETANDO LINHAS E AUMENTANDO PÁGS USADAS, CASO NÃO TENHA MAIS ESPAÇO PARA LINHAS NA PÁGINA
        if (countLinhas >= L){
            countLinhas = 0;
            countPags++;
        }
        
    }
    printf("%d",countPags);
    return 0;
 }

File: BeeCrowd_Python/1222.py
# N = Número de palavras
# L = Número de linhas 
# C = Número de caracteres por linha

def main():
    while True:
        entrada = input().split()
        n_palavras = input()
        N = int(entrada[0])
        L = int(entrada[1])
        C = int(entrada[2])
        novo_texto = ''
        count_caracteres = 0
        # O LIVRO JÁ COMEÇARIA AUTOMATICAMENTE NA PRIMEIRA LINHA
        count_linhas = 1
        count_pags = 1
        
        for letra in n_palavras:
            # VERIFICANDO SE AINDA CABEM CARACTERES NA LINHA
            if count_caracteres < C:
                count_caracteres += 1
            # RESETANDO CARACTERES E AUMENTANDO LINHAS USADAS, CASO NÃO TENHA MAIS ESPAÇO PARA CARACTERES NA LINHA
            elif count_caracteres >= C:
                count_caracteres = 0
                count_linhas += 1
            # RESETANDO LINHAS E AUMENTANDO PÁGS USADAS, CASO NÃO TENHA MAIS ESPAÇO PARA LINHAS NA PÁGINA
            if count_linhas == L:
                count_pags += 1
                count_linhas = 0
        
        print(count_pags)
main()